ec/src/lib.rs

```diff
@@ -1,11 +1,5 @@
 #![cfg_attr(not(feature = "std"), no_std)]
-#![deny(
-    warnings,
-    unused,
-    future_incompatible,
-    nonstandard_style,
-    rust_2018_idioms
-)]
+#![warn(future_incompatible, nonstandard_style, rust_2018_idioms)]
```

Comment:
Is the `unused` warn by default? Otherwise, we should keep it here.

Reply:
A couple of the unused warns are allowed by default (namely 
```
unused_crate_dependencies
unused_extern_crates
unused_import_braces
unused_lifetimes
unused_qualifications
unused_results
```
)
I'll `unused` back to warn